plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/StateConfVectorIndexCalculator.xtend

```diff
@@ -0,0 +1,46 @@
+package org.yakindu.sct.generator.c
+
+import com.google.inject.Inject
+import java.util.HashMap
+import java.util.Map
+import org.yakindu.sct.model.sexec.ExecutionState
+import org.yakindu.sct.model.sexec.naming.INamingService
+
+class StateConfVectorIndexCalculator {
+	@Inject extension INamingService
+	protected Map<ExecutionState, Integer> indexMap
+	protected Map<ExecutionState, String> defineMap
+	
+	def Integer getIndex(ExecutionState it) {
```

Comment:
I agree!
- The class does not calculate anything as the calculation of the index is already calculated.
- it is effectively just a shortcut access extension that allows to write `it.index` instead of `it.stateVector.offset`.
- `it.stateVector.offset` is more precise about what is really meant.

So - please remove this class